rust/eos-rs/src/api/dungeon_mode/entity.rs

```diff
@@ -1,69 +1,40 @@
+/// These implementations can assume that overlay 29 is loaded (since this is the
+/// only context dungeon entities are actually relevant). If you somehow manually construct types
+/// of this struct for some reason, you NEED to make sure overlay 29 is loaded when using some f
+/// unctions of this trait.
+/// The methods in this impl are marked safe for convenience, since in its intended use case,
 /// overlay 29 will always be loaded.
-pub trait DungeonEntityExt {
+impl DungeonEntity {
     /// Checks if a given entity is actually valid.
-    fn is_valid(slf: *mut Self) -> bool;
-
-    /// Entity type. Invalid values will return None.
-    fn entity_type(&self) -> Option<DungeonEntityType>;
-
-    /// This returns the monster info struct for the entity,
-    /// panics if the entity is not a monster.
-    fn info_for_monster(&self) -> Option<DungeonMonsterRef>;
-
-    /// This returns the item info struct for the entity,
-    /// panics if the entity is not an item.
-    fn info_for_item(&self) -> Option<&DungeonItem>;
-
-    /// This returns the trap info struct for the entity,
-    /// panics if the entity is not a trap.
-    fn info_for_trap(&self) -> Option<&DungeonTrap>;
-
-    /// This returns the monster info struct for the entity,
-    /// panics if the entity is not a monster.
-    fn info_for_monster_mut(&mut self) -> Option<DungeonMonsterMut>;
-
-    /// This returns the item info struct for the entity,
-    /// panics if the entity is not an item.
-    fn info_for_item_mut(&mut self) -> Option<&mut DungeonItem>;
-
-    /// This returns the trap info struct for the entity,
-    /// panics if the entity is not a trap.
-    fn info_for_trap_mut(&mut self) -> Option<&mut DungeonTrap>;
-
-    /// Returns the tile, that the entity is located at.
-    fn get_tile(&self) -> Option<&DungeonTile>;
-
-    /// Returns the tile, that the entity is located at.
-    fn get_tile_mut(&mut self) -> Option<&mut DungeonTile>;
-}
-
-impl DungeonEntityExt for DungeonEntity {
     #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    fn is_valid(slf: *mut Self) -> bool {
+    pub fn is_valid(slf: *mut Self) -> bool {
         // SAFETY: The lease passed into the function promises us that the overlay is loaded.
         //         Since this function is intended to actually check if the entity is valid,
         //         it's safe to call, even if the `slf` pointer is invalid.
         unsafe { ffi::EntityIsValid(slf) > 0 }
     }
 
-    fn entity_type(&self) -> Option<DungeonEntityType> {
+    /// Entity type. Invalid values will return None.
+    pub fn entity_type(&self) -> Option<DungeonEntityType> {
         self.type_.try_into().ok()
     }
 
-    fn info_for_monster(&self) -> Option<DungeonMonsterRef> {
+    /// This returns the monster info struct for the entity,
+    /// panics if the entity is not a monster.
+    pub fn info_for_monster(&self) -> Option<DungeonMonsterRef> {
```

Comment:
Would it make more sense to implement `TryFrom<&DungeonEntity> for DungeonMonsterRef`/`TryFrom<&mut DungeonEntity> for DungeonMonsterMut`, etc.? I'm not sure how the ergonomics would work out.

Reply:
I was considering that, but from what I've seen other libs do I feel like having trait implementations for references is usually not the way to go since it's not very obvious.